lib/SDIR/Translate/translateToSDFG.cpp

```diff
@@ -0,0 +1,1597 @@
+    }
+  jemit.endObject(); // _arrays
+
+  jemit.startNamedObject("symbols");
+  for (Operation &oper : op.body().getOps()) {
+    if (AllocSymbolOp alloc = dyn_cast<AllocSymbolOp>(oper))
+      if (translateToSDFG(*alloc, jemit).failed())
+        return failure();
+  }
+
+  jemit.endObject(); // symbols
+
+  if (!containsAttr(*op, "instrument"))
+    jemit.printKVPair("instrument", "No_Instrumentation");
+
+  jemit.startNamedObject("init_code");
+  jemit.startNamedObject("frame");
+  jemit.printKVPair("string_data", "");
+  jemit.printKVPair("language", "CPP");
+  jemit.endObject(); // frame
+  jemit.endObject(); // init_code
+
+  jemit.startNamedObject("exit_code");
+  jemit.startNamedObject("frame");
+  jemit.printKVPair("string_data", "");
+  jemit.printKVPair("language", "CPP");
+  jemit.endObject(); // frame
+  jemit.endObject(); // exit_code
+
+  jemit.startNamedObject("global_code");
+  jemit.startNamedObject("frame");
+  jemit.printKVPair("string_data", "");
+  jemit.printKVPair("language", "CPP");
+  jemit.endObject(); // frame
+  jemit.endObject(); // global_code
+
+  jemit.printKVPair("name", op.sym_name());
+  jemit.endObject(); // attributes
+
+  jemit.startNamedList("nodes");
+
+  unsigned stateID = 0;
+
+  for (Operation &oper : op.body().getOps())
+    if (StateNode state = dyn_cast<StateNode>(oper)) {
+      state.setID(stateID);
+      if (translateStateToSDFG(state, jemit).failed())
+        return failure();
+      stateID++;
+    }
+
+  jemit.endList(); // nodes
+
+  jemit.startNamedList("edges");
+
+  for (Operation &oper : op.body().getOps())
+    if (EdgeOp edge = dyn_cast<EdgeOp>(oper))
+      if (translateToSDFG(*edge, jemit).failed())
+        return failure();
+
+  jemit.endList(); // edges
+
+  StateNode entryState = op.getStateBySymRef(op.entry());
+  jemit.printKVPair("start_state", entryState.ID(), /*stringify=*/false);
+
+  return success();
+}
+
+LogicalResult translateSDFGToSDFG(SDFGNode &op, JsonEmitter &jemit) {
+  if (!op.isNested()) {
+    jemit.startObject();
+
+    if (printSDFGNode(op, jemit).failed())
+      return failure();
+
+    jemit.endObject();
+    return success();
+  }
+
+  jemit.startObject();
+  jemit.printKVPair("type", "NestedSDFG");
+  jemit.printKVPair("id", op.ID(), /*stringify=*/false);
+
+  jemit.startNamedObject("attributes");
+
+  if (!containsAttr(*op, "instrument"))
+    jemit.printKVPair("instrument", "No_Instrumentation");
+
+  if (!containsAttr(*op, "schedule"))
+    jemit.printKVPair("schedule", "Default");
+
+  jemit.startNamedObject("in_connectors");
+  for (BlockArgument bArg : op.getArguments()) {
+    AsmState state(op);
+    std::string name;
+    llvm::raw_string_ostream nameStream(name);
+    bArg.printAsOperand(nameStream, state);
+    name.erase(0, 1); // Remove %-sign
+    jemit.printKVPair(name, "null", /*stringify=*/false);
+  }
+  jemit.endObject(); // in_connectors
+
+  jemit.startNamedObject("out_connectors");
+  // TODO: Print out_connectors
```

Comment:
The assumption of only a `__return` output is problematic here, isn't it?

Reply:
Yes indeed. For now the translator only supports tasklets that return a single value. This is fine for all the evaluation we'd like to do, but should be changed to account for the output names attribute.